object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedObjectStore.java

```diff
@@ -224,9 +235,9 @@ private void tryReportUpdate(
         configChangeEventGenerator ->
             configChangeEventGenerator.sendUpdateNotification(
                 requestContext,
-                result.getData().getClass().getName(),
+                this.buildClassNameForChangeEvent(result.getData()),
                 result.getContext(),
-                previousValue,
-                this.buildValueFromData(result.getData())));
+                this.buildDataFromValue(previousValue).map(this::buildValueForChangeEvent).get(),
```

Comment:
>  Not sure how can we ensure that aspect.

We can't, but the consumer now has the option to react how they deem appropriate. They're always in charge of deserialization and need to handle any issues anyway, since there's no guarantee that the deserialization, even if successful here, matches what that consumer is doing.

Reply:
I know it is not a concern with current consumers as we are well aware of the scenarios. It is sort of tribal knowledge, which should not be the case.

Reply:
> We can't, but the consumer now has the option to react how they deem appropriate. They're always in charge of deserialization and need to handle any issues anyway, since there's no guarantee that the deserialization, even if successful here, matches what that consumer is doing.

Makes sense. Let's go with your suggestion.